### PR TITLE
Document behavior of daemon log options with respect to existing containers.

### PR DESCRIPTION
**Pull request by colinmollenhour**
### Proposed changes

Add notes in docs as suggested by @thaJeztah in docker/docker#31975.

### CODE REVIEW

File: _data/dockerd-cli/dockerd.yaml [hunk 1]
```diff
@@ -1182,7 +1182,15 @@ long: |
...
 
-
+  ### Configuration inheritance
+  
+  Some options will not take effect on existing containers and will only take effect on new containers. These include
+  options that override defaults for options used by `docker run` such as:
+  
+  - `log-driver`
+  - `log-opts`
+  - Potentially others...
```
Comment (mstanleyjones) [inline _data/dockerd-cli/dockerd.yaml]: Potentially others? Can we narrow this down?

File: engine/admin/logging/overview.md [hunk 2]
```diff
@@ -32,6 +32,10 @@ $ docker info |grep 'Logging Driver'
 Logging Driver: json-file
 ```
 
+If the default logging driver is changed via the Docker daemon command line flags `--log-driver`
+and `--log-opt` or `daemon.json` options `log-driver` and `log-opts`, the new 
+defaults will  not be applied to existing containers, only new containers.
```
Comment (mstanleyjones) [inline engine/admin/logging/overview.md]: nit: extra space after "will"

File: _data/dockerd-cli/dockerd.yaml (hunk 1 above)
Comment (colinmollenhour) [inline _data/dockerd-cli/dockerd.yaml]: I read through all of the options but it was not apparent to me so I think someone with better knowledge of internals would be better able to do this.

Comment (mstanleyjones) [inline _data/dockerd-cli/dockerd.yaml]: @vdemeester @thaJeztah would one of you know who can answer this?

**Comment by colinmollenhour** [comment]
Just FYI I've done all I'm going to do on this PR.

**Comment by mstanleyjones** [comment]
I'm not prepared to merge this until we get clarification on the "potentially others". No harm in leaving it open a little longer.

**Comment by mstanleyjones** [comment]
Any news on this? I'm still not happy with the vagueness of the advice.

**Comment by mstanleyjones** [comment]
Out of date.

**Comment by colinmollenhour** [comment]
Are you saying this is no longer relevant or just no activity?

**Comment by mstanleyjones** [comment]
Well, you had "done all you were going to do" and it was not ready to be merged.

**Comment by colinmollenhour** [comment]
Ok, just clarifying if something was updated in a recent release or if you were really going to reject a documentation PR which is a net improvement despite it not being perfect. :)

**Pull request closed without merging**